src/map_extras.cpp

```diff
@@ -76,6 +76,164 @@ void mx_null( map &, const tripoint & )
+                             bbox.p1.y );
+
+    int x_offset = veh->dir_vec().x * ( x_length / 2 );   // cont.
+    int y_offset = veh->dir_vec().y * ( y_length / 2 );
+
+    int x_min = abs( bbox.p1.x ) + 0;
+    int y_min = abs( bbox.p1.y ) + 0;
+
+    int x_max = SEEX * 2 - bbox.p2.x + 1;
+    int y_max = SEEY * 2 - bbox.p2.y + 1;
+
+    int x1 = clamp( cx + x_offset, x_min,
+                    x_max );
+    int y1 = clamp( cy + y_offset, y_min, y_max ); 
+    // Clamp x1 & y1 such that no parts of the vehicle extend
+    // over the border of the submap.
+
+    vehicle *wreckage = m.add_vehicle( city_wrecks, tripoint( x1, y1, abs_sub.z ), dir1, rng( 1, 33 ),
+                                       1 );
+
+    const auto controls_at = []( vehicle * wreckage, const tripoint & pos ) {
+        return !wreckage->get_parts_at( pos, "CONTROLS", part_status_flag::any ).empty() ||
+               !wreckage->get_parts_at( pos, "CTRL_DASHBOARD", part_status_flag::any ).empty():
+    };
+
+    if( wreckage != nullptr ) {
+        const int clowncar_factor = dice( 1, 8 );
+
+        switch( clowncar_factor ) {
+            case 1:
+            case 2:
+            case 3: // Full clown car
+                for( const vpart_reference &vp : wreckage->get_any_parts( VPFLAG_SEATBELT ) ) {
+                    const tripoint pos = vp.pos();
+                    // Spawn pilots in seats with controls.CTRL_ELECTRONIC
+                    if( controls_at( wreckage, pos ) ) {
+                        m.add_spawn( mon_zombie, 1, pos.x, pos.y );
+                    } else {
+                        if( one_in( 5 ) ) {
+                            m.add_spawn( mon_zombie, 1, pos.x, pos.y );
+                    }
+                }
+
+                    // Delete the items that would have spawned here from a "corpse"
+                    wreckage->get_items( sp ).clear();
+                        }
+                    }
+                }
+                break;
+            case 4:
+            case 5: // 2/3rds clown car
+                for( const vpart_reference &vp : wreckage->get_any_parts( VPFLAG_SEATBELT ) ) {
+                    const tripoint pos = vp.pos();
+                    // Spawn pilots in seats with controls.
+                    if( controls_at( wreckage, pos ) ) {
+                        m.add_spawn( mon_zombie, 1, pos.x, pos.y );
+                    } else {
+                        if( !one_in( 3 ) ) {
+                            m.add_spawn( mon_zombie, 1, pos.x, pos.y );
+                        }
+                    }
+
+                    // Delete the items that would have spawned here from a "corpse"
+                    for( auto sp : wreckage->parts_at_relative( vp.mount(), true ) ) {
+                        vehicle_stack here = wreckage->get_items( sp );
+
+                        for( auto iter = here.begin(); iter != here.end(); ) {
```

Comment:
Here you can use the same trick as above: calling `clear` instead of the loop. Applies to the loops below as well.